sample-code/vc-using-framestream/vc-using-framestream/ArisRecording.cpp

```diff
@@ -0,0 +1,154 @@
+#include "ArisRecording.h"
+#include "FrameHeader.h"
+#include "FileHeader.h"
+#include "Reorder.h"
+#include <assert.h>
+#include <functional>
+
+std::unique_ptr<WriteableArisRecording> WriteableArisRecording::Create(const char * path)
+{
+  try {
+    // Overwrite the file if it exists.
+    std::ofstream outfile(path, std::ios::binary | std::ios::out | std::ios::trunc);
+    if (outfile) {
+      if (WriteFileHeader(outfile)) {
+        return std::make_unique<WriteableArisRecording>(path, std::move(outfile));
+      }
+
+      // No reason to continue if we can't write the file header.
+      outfile.close();
+      std::remove(path);
+
+      // Fall through.
+    }
+  }
+  catch (...) {
+    // Fall through.
+  }
+
+  // Unassociated ptr == failure.
+  return std::unique_ptr<WriteableArisRecording>();
+}
+
+WriteableArisRecording::WriteableArisRecording(const char * path, std::ofstream && initializedOutfile)
+  : path_(path)
+  , outfile_(std::move(initializedOutfile))
+  , frameCount_(0)
+{
+}
+
+WriteableArisRecording::~WriteableArisRecording()
+{
+  if (outfile_.is_open()) {
+    outfile_.seekp(0, std::ios_base::end);
+    const bool hasFrames = outfile_.tellp() > sizeof ArisFileHeader;
+    outfile_.close();
+
+    // If it's an empty file, remove it.
+    if (!hasFrames) {
+      std::remove(path_.c_str());
+    }
+  }
+}
+
+/* static */
+bool WriteableArisRecording::WriteFileHeader(std::ofstream & outfile)
+{
+  ArisFileHeader header;
+  memset(&header, 0, sizeof header);
+  header.Version = ARIS_FILE_SIGNATURE;
+
+  assert(outfile.tellp() == 0);
+
+  outfile.write(reinterpret_cast<const char*>(&header), sizeof header);
```

Comment:
Would rather fix the `write()` signature in stdlib. For some reason they decided it should be 'characters'.